x-pack/legacy/plugins/index_management/__jest__/client_integration/helpers/setup_environment.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import axios from 'axios';
+import axiosXhrAdapter from 'axios/lib/adapters/xhr';
+import { init as initHttpRequests } from './http_requests';
+import { AppContextProvider } from '../../../public/application/app_context';
+import { httpService } from '../../../public/application/services/http';
+import { breadcrumbService } from '../../../public/application/services/breadcrumbs';
+import { documentationService } from '../../../public/application/services/documentation';
+import { notificationService } from '../../../public/application/services/notification';
+import { ExtensionsService } from '../../../public/services';
+import { UiMetricService } from '../../../public/application/services/ui_metric';
+import { setUiMetricService } from '../../../public/application/services/api';
+import { setExtensionsService } from '../../../public/application/store/selectors';
+
+/* eslint-disable @kbn/eslint/no-restricted-paths */
+import { notificationServiceMock } from '../../../../../../../src/core/public/notifications/notifications_service.mock';
+import { docLinksServiceMock } from '../../../../../../../src/core/public/doc_links/doc_links_service.mock';
+
+const mockHttpClient = axios.create({ adapter: axiosXhrAdapter });
+
+export const services = {
+  extensionsService: new ExtensionsService(),
+  uiMetricService: new UiMetricService('index_management'),
+};
+services.uiMetricService.setup({ reportUiStats() {} } as any);
+setExtensionsService(services.extensionsService);
+setUiMetricService(services.uiMetricService);
+const appDependencies = { services, core: {}, plugins: {} } as any;
+
+export const setupEnvironment = () => {
+  // Mock initialization of services
+  // @ts-ignore
+  httpService.setup(mockHttpClient);
```

Comment:
can you use the `http_service.mock` now?

Reply:
Thanks for the suggestion. I ended up trying with 

```js
// httpServiceMock.createSetupContract() returns an HttpSetupMock from core
httpService.setup(httpServiceMock.createSetupContract());
```

but this breaks our tests. Probably because our `sinon.fakeServer() requires a real http client like `axios`. Will need to be investigated and changed in all our apps later.

Reply:
FWIW I did get this working for `remote_clusters` (PR: https://github.com/elastic/kibana/pull/57365). I did have to adjust the timeouts though.

Reply:
Cool. Did you try by providing directly the `httpServiceMock.createSetupContract()) ` from `core`?